image/descriptor.go

```diff
@@ -49,6 +49,30 @@ func (d *descriptor) hash() string {
 	return pts[1]
 }
 
+func findDescriptors(w walker) (map[string]*descriptor, error) {
+	refs := make(map[string]*descriptor)
+
+	if err := w.walk(func(path string, info os.FileInfo, r io.Reader) error {
+		if info.IsDir() || !strings.HasPrefix(path, "refs") {
+			return nil
+		}
+
+		var d descriptor
+		if err := json.NewDecoder(r).Decode(&d); err != nil {
+			return err
+		}
+		refs[info.Name()] = &d
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	if len(refs) == 0 {
+		return nil, errors.New("no descriptors found")
```

Comment:
Is this a problem?  I don't think `findDescriptors` ([`listReferences`?]) should error if there aren't any references.  It should just return an empty map.

I think `validate` should *also* succeed if there are no refs in the store and the caller does not ask for any specifically.  There are still no invalid refs in the store.  However, if we want to special-case validate to error out if the caller sets and empty `refs` and the store has no refs, I'm ok with that (in which case we'd move this check over to `validate`).

[1]: https://github.com/opencontainers/image-spec/pull/279/files#r78029573